### PR TITLE
Linting

### LINKED ISSUE #3: Lint the fuck out of it
This codebase is full to the brim of lint issues. We should get it cleaned up.

### PR DESCRIPTION
**Pull request by harry1453**
Closes #3 

Linting performed using IntelliJ IDEA default linter with default options. Number of warnings is the number of normal or greater severity warnings (not including spelling warnings, weak warnings or code duplication warnings)

Before - 2060 warnings

After - 1250 warnings, 842 of which are unchecked call / assignment warnings from the compiler, as throughout the codebase generics are not explicitly typed. This would take a lot of time and would have to be done carefully, so I am happy with a ~40% decrease in warnings for now.

### CODE REVIEW

**Comment by blankey1337** [comment]
Very cool! Can we resolve #5 before merging this so master and develop don't diverge too early?

File: src/brs/db/sql/SqlAccountStore.java [hunk 1]
```diff
@@ -147,31 +143,23 @@ protected void bulkInsert(DSLContext ctx, ArrayList<Account> accounts) {
...
+                DbKey dbKey = (DbKey) accountDbKeyFactory.newKey(account.getId());
                 getCache().put(dbKey, new SqlAccount(account.intoResultSet()));
-              }
-              catch ( SQLException e ) {
+              } catch (SQLException e) {
                 // ignore
               }
             }
           }
-          finally {
```
Comment (ohager) [inline src/brs/db/sql/SqlAccountStore.java]: good catch

File: src/brs/db/sql/VersionedBatchEntitySqlTable.java [hunk 2]
```diff
@@ -81,12 +77,10 @@ public void finish() {
...
 
       BatchBindStep updateBatch = ctx.batch(updateQuery);
-      Iterator<DbKey> it = keySet.iterator();
-      while (it.hasNext()) {
-        DbKey dbKey = it.next();
+      for (DbKey dbKey : (Iterable<DbKey>) keySet) {
         ArrayList<Object> bindArgs = new ArrayList<>();
         bindArgs.add(false);
-        Arrays.stream(dbKey.getPKValues()).forEach(pkValue -> bindArgs.add(pkValue));
+        Arrays.stream(dbKey.getPKValues()).forEach(bindArgs::add);
```
Comment (ohager) [inline src/brs/db/sql/VersionedBatchEntitySqlTable.java]: :heart:

File: src/brs/http/API.java [hunk 3]
```diff
@@ -166,19 +168,16 @@ public API(TransactionProcessor transactionProcessor,
...
-              apiServer.start();
-              logger.info("Started API server at " + host + ":" + port);
-            } catch (Exception e) {
-              logger.error("Failed to start API server", e);
-              throw new RuntimeException(e.toString(), e);
-            }
-
-          }
-        }, true);
+      threadPool.runBeforeStart(() -> {
```
Comment (ohager) [inline src/brs/http/API.java]: :smile:

File: src/brs/http/APITestServlet.java [hunk 4]
```diff
@@ -55,7 +54,7 @@
       + "            }\n"
       + "            $.ajax({\n"
       + "                url: url,\n"
-      + "                type: 'GET',\n"
+      + "                type: 'POST',\n"
```
Comment (ohager) [inline src/brs/http/APITestServlet.java]: nice, although I ask myself, what's the value of this header string...

File: src/brs/http/GetMyPeerInfo.java [hunk 5]
```diff
@@ -2,17 +2,18 @@
...
+import javax.servlet.http.HttpServletRequest;
+
+final class GetMyPeerInfo extends APIServlet.APIRequestHandler {
 
   private final BlockchainProcessor blockchainProcessor;
   private final TransactionProcessor transactionProcessor;
 
   public GetMyPeerInfo(BlockchainProcessor blockchainProcessor, TransactionProcessor transactionProcessor) {
-    super(new APITag[]{APITag.INFO});
+    super(new APITag[]{APITag.PEER_INFO});
```
Comment (ohager) [inline src/brs/http/GetMyPeerInfo.java]: a fix? if so, nice catch

File: src/brs/util/ThreadPool.java [hunk 6]
```diff
@@ -114,13 +114,10 @@ public synchronized void start(int timeMultiplier) {
...
 
     logger.debug("Starting " + afterStartJobs.size() + " delayed tasks");
-    Thread thread = new Thread() {
-        @Override
-        public void run() {
-          runAll(afterStartJobs);
-          afterStartJobs.clear();
-        }
-      };
+    Thread thread = new Thread(() -> {
```
Comment (ohager) [inline src/brs/util/ThreadPool.java]: :+1:

File: src/brs/GeneratorImpl.java [hunk 7]
```diff
@@ -172,7 +167,7 @@ public BigInteger calculateDeadline(long accountId, long nonce, byte[] genSig, i
     private final Long accountId;
     private final String secretPhrase;
     private final byte[] publicKey;
-    private volatile BigInteger deadline;
+    private final BigInteger deadline;
```
[the remark this answers is not in the document]
Comment (harry1453) [reply src/brs/GeneratorImpl.java]: yeah, that's weird. didn't notice that one, we should keep this in mind.

**Comment by harry1453** [comment]
Note: this also contains some stuff that's about to be merged into develop from master so it MUST NOT be squashed.

**Pull request merged**